src/main/scala/org/hammerlab/guacamole/readsets/rdd/RegionRDD.scala

```diff
@@ -0,0 +1,192 @@
+   *
+   * Useful for getting a sense of which parts of the genome have exceedingly high coverage.
+   */
+  def partitionDepths(halfWindowSize: Int,
+                      loci: LociSet,
+                      depthCutoff: Int): RDD[((ContigName, Boolean), Int)] = {
+    (for {
+      (Position(contig, _), Coverage(depth, _, _)) <- coverage(halfWindowSize, loci)
+    } yield
+      contig -> (depth <= depthCutoff)
+      ).runLengthEncode
+  }
+
+  /**
+   * Alternative implementation of this.coverage; will generally perform significantly worse on sorted inputs. Useful as
+   * a sanity check.
+   */
+  def shuffleCoverage(halfWindowSize: Int, contigLengthsBroadcast: Broadcast[ContigLengths]): RDD[PositionCoverage] = {
+    rdd
+    .flatMap(r => {
+      val c = r.contigName
+      val length = contigLengthsBroadcast.value(c)
+
+      val lowerBound = math.max(0, r.start - halfWindowSize)
+      val upperBound = math.min(length, r.end + halfWindowSize)
+
+      val outs = ArrayBuffer[(Position, Coverage)]()
+      for {
+        l <- lowerBound until upperBound
+      } {
+        outs += Position(c, l) -> Coverage(depth = 1)
+      }
+
+      outs += Position(c, lowerBound) -> Coverage(starts = 1)
+      outs += Position(c, upperBound) -> Coverage(ends = 1)
+
+      outs.iterator
+    })
+    .reduceByKey(_ + _)
+    .sortByKey()
+  }
+
+  /**
+   * Wrappers around this.coverage that map genomic loci to their depth and numbers of read starts and ends, sorted in
+   * descending order of the latter.
+   */
+  @transient val depths_ = mutable.Map[(Int, LociSet), RDD[(Int, Position)]]()
+  def depths(halfWindowSize: Int, loci: LociSet): RDD[(Int, Position)] =
+    depths_.getOrElseUpdate(
+      halfWindowSize -> loci,
+      coverage(halfWindowSize, loci).map(t => t._2.depth -> t._1).sortByKey(ascending = false)
+    )
+
+  @transient val starts_ = mutable.Map[(Int, LociSet), RDD[(Int, Position)]]()
+  def starts(halfWindowSize: Int, loci: LociSet): RDD[(Int, Position)] =
+    starts_.getOrElseUpdate(
+      halfWindowSize -> loci,
+      coverage(halfWindowSize, loci).map(t => t._2.starts -> t._1).sortByKey(ascending = false)
+    )
+
+  @transient val ends_ = mutable.Map[(Int, LociSet), RDD[(Int, Position)]]()
+  def ends(halfWindowSize: Int, loci: LociSet): RDD[(Int, Position)] =
+    ends_.getOrElseUpdate(
+      halfWindowSize -> loci,
+      coverage(halfWindowSize, loci).map(t => t._2.ends -> t._1).sortByKey(ascending = false)
+    )
+}
+
+object RegionRDD {
+  private val rddMap = mutable.Map[Int, RegionRDD[_]]()
+  implicit def rddToRegionRDD[R <: ReferenceRegion: ClassTag](rdd: RDD[R]): RegionRDD[R] =
+    rddMap.getOrElseUpdate(
+      rdd.id,
+      new RegionRDD[R](rdd)
+    ).asInstanceOf[RegionRDD[R]]
+
+  /**
+   * Given an RDD output by {{RegionRDD.partitionDepths}}, count the number of loci below (resp. above) the depth
```

Comment:
I think you mean "keyed by" in `Given an RDD output by`? If you could be a little more explicit in the comment on what `depthRuns` contains that'd be helpful